lib/agent.js

```diff
@@ -450,18 +430,58 @@ Agent.prototype.startAggregators = function startAggregators() {
   }
 }
 
-Agent.prototype.onConnect = function onConnect() {
-  this.metrics.reconfigure(this.config)
-  this.errors.reconfigure(this.config)
-  this.traces.reconfigure(this.config)
-  this.queries.reconfigure(this.config)
-  this.spanEventAggregator.reconfigure(this.config)
-  this.transactionEventAggregator.reconfigure(this.config)
-  this.customEventAggregator.reconfigure(this.config)
+/**
+ * Completes any final setup upon full connection to New Relic
+ * servers and sets the agent state to 'started'.
+ */
+Agent.prototype.onConnect = function onConnect(shouldImmediatelyHarvest, callback) {
+  this._reconfigureAggregators(this.config)
 
   if (this.config.certificates && this.config.certificates.length > 0) {
     this.metrics.getOrCreateMetric(NAMES.FEATURES.CERTIFICATES).incrementCallCount()
   }
+
+  this._scheduleHarvests(shouldImmediatelyHarvest, callback)
+
+  this.setState('started')
+}
```

Comment:
onConnect now handles consistency of setting up aggregators and starting them, and setting agent state to 'started.'

Reply:
Given there's now always an async hop, moving `setState` to after the harvest scheduling should have no impact. But mentioning since that was a change. I don't see anything in the starting of the aggregators, etc. that relies on that already being set.